cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -476,75 +476,74 @@ func TestDrain(t *testing.T) {
 			pdbs:        []*policyv1.PodDisruptionBudget{emptyPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&rc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{rcPod},
 		},
 		{
 			description: "kube-system PDB with matching kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{kubeSystemRcPod},
 		},
 		{
 			description: "kube-system PDB with non-matching kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemFakePDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: true,
 			expectPods:  []*apiv1.Pod{},
 		},
 		{
 			description: "kube-system PDB with default namespace pod",
 			pods:        []*apiv1.Pod{rcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&rc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{rcPod},
 		},
 		{
 			description: "default namespace PDB with matching labels kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{defaultNamespacePDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: true,
 			expectPods:  []*apiv1.Pod{},
 		},
 	}
 
 	for _, test := range tests {
-		fakeClient := &fake.Clientset{}
-		register := func(resource string, obj runtime.Object, meta metav1.ObjectMeta) {
-			fakeClient.Fake.AddReactor("get", resource, func(action core.Action) (bool, runtime.Object, error) {
-				getAction := action.(core.GetAction)
-				if getAction.GetName() == meta.GetName() && getAction.GetNamespace() == meta.GetNamespace() {
-					return true, obj, nil
-				}
-				return false, nil, fmt.Errorf("Not found")
-			})
-		}
+		var err error
+		var rcLister v1lister.ReplicationControllerLister
 		if len(test.rcs) > 0 {
-			register("replicationcontrollers", &test.rcs[0], test.rcs[0].ObjectMeta)
+			rcLister, err = kube_util.NewTestReplicationControllerLister(test.rcs)
+			assert.NoError(t, err)
 		}
-		register("daemonsets", &ds, ds.ObjectMeta)
-		register("jobs", &job, job.ObjectMeta)
-		// register("statefulsets", &statefulset, statefulset.ObjectMeta)
-
+		var rsLister v1appslister.ReplicaSetLister
 		if len(test.replicaSets) > 0 {
-			register("replicasets", &test.replicaSets[0], test.replicaSets[0].ObjectMeta)
+			rsLister, err = kube_util.NewTestReplicaSetLister(test.replicaSets)
+			assert.NoError(t, err)
 		}
+
+		dsLister, err := kube_util.NewTestDaemonSetLister([]*appsv1.DaemonSet{&ds})
```

Comment:
Nit: consider changing interface of NewTest*Listers so they take argument list of variable length instead of array.